Bump FrontbaseNIO to 1.6.1.

tancred opened the pull request:
Not sure why the test failed for me, but the coding path count wasn't zero anymore (or perhaps "anymore" is the wrong word and the test should've been update a while back?).

Aaaanyway. 😅

tancred committed:
Bump FrontbaseNIO to 1.6.1.

johan-carlberg commented:
When I run the tests, the coding path count is still zero, so something is going on here. I guess we need to investigate this further.

tancred commented:
🤔

Is there a problem with versions? I run the currently latest version 8.2.33 on macOS (trying to test on linux in a container too, but fails miserably at the moment; can't get the tests to complete and have a feeling it has to do with event loop groups, thread pools and the likes).

johan-carlberg commented:
I've occasionally seen errors where the table already existed, so there seems to be some issue with setup/teardown. I noticed that the tests use a networked database with a fixed name, so it might be the case that tests are using the same database simultaneously.

johan-carlberg commented:
I noticed that the `codingPath` in the error being tested is specifically set to the empty array, so the count should really be zero.

tancred commented:
> I noticed that the tests use a networked database with a fixed name, so it might be the case that tests are using the same database simultaneously.

Right! I had to start FBExec when I ran the tests in a container, and I also noted that the "template" db name construction doesn't seem to work. It doesn't seem to replace the XXX... with something unique.

johan-carlberg commented:
> Right! I had to start FBExec when I ran the tests in a container, and I also noted that the "template" db name construction doesn't seem to work. It doesn't seem to replace the XXX... with something unique.

I don't know why we're using networked databases for the tests here. Perhaps because there were issues with multiple connections to file based databases, but since that was resolved in `FrontbaseNIO` 1.6.0 we might switch to file based databases in the tests.

tancred committed:
Use file-based database for testing.

tancred commented:
Tried switching to a file-based database for testing. Now the tests pass for me on linux (in a container running debian 12, FrontBase 8.2.32 and Swift 6).

tancred commented:
> Tried switching to a file-based database for testing. Now the tests pass for me on linux (in a container running debian 12, FrontBase 8.2.32 and Swift 6).

...and on macOS with FrontBase 8.2.33.

johan-carlberg commented:
It still fails for me
```
Tests/FrontbaseTests/FrontbaseTests.swift:163: error: -[FrontbaseTests.FrontbaseTests testDecodingNull] : XCTAssertEqual failed: ("0") is not equal to ("1")
Swift/ContiguousArrayBuffer.swift:675: Fatal error: Index out of range
```

I'm on macOS with FrontBase 8.2.33.

tancred commented:
> It still fails for me
> 
> ```
> Tests/FrontbaseTests/FrontbaseTests.swift:163: error: -[FrontbaseTests.FrontbaseTests testDecodingNull] : XCTAssertEqual failed: ("0") is not equal to ("1")
> Swift/ContiguousArrayBuffer.swift:675: Fatal error: Index out of range
> ```
> 
> I'm on macOS with FrontBase 8.2.33.

Wat!?

Hmm, and you're not running into cached dependencies/packages somehow?

I realized I build with dynamic linking, so I switched to static. But, that didn't take effect until I deleted a lot of cached stuff. I ended up deleting `.build`, `.swiftpm`, `~/Library/org.swift.swiftpm` and `~/Library/Caches/org.swift.swiftpm` and totally wiped the Xcode build folder down in `~/Library/Developer/Xcode/DerivedData/Frontbase-...` (I'm pretty sure I didn't have to delete all of those to get the linking change to "stick", but I haven't had time to check exactely which ones are needed).

johan-carlberg commented:
Emptying the caches makes no difference.

I cannot understand how you get anything in `context.codingPath`, since the error thrown explicitly have an empty array as `codingPath`.

```
            let context = DecodingError.Context (codingPath: [], debugDescription: "Value not convertible (\(column))")
            throw DecodingError.typeMismatch (type, context)
```

tancred commented:
Hmm. When I step through I end up in [SQLKit's SQLCodingUtilities.swift](https://github.com/vapor/sql-kit/blob/e0b35ff07601465dd9f3af19a1c23083acaae3bd/Sources/SQLKit/Rows/SQLCodingUtilities.swift#L133) where the context is augmented with the coding path.

Are we using the same version of SQLKit? In my checkout I've version 3.32.0 checked out.

johan-carlberg commented:
I had SQLKit 3.28.0.
After updating to 3.32.0, everything works.

johan-carlberg reviewed (approved):
Nice!
